astroquery/esasky/tests/test_esasky_remote.py

```diff
@@ -1,352 +1,281 @@
         assert "1342246640" in result["HERSCHEL"].columns["observation_id"]
 
-    def test_esasky_get_images_obs_id(self):
-        download_directory = "ESASkyRemoteTest"
-        if not os.path.exists(download_directory):
-            os.makedirs(download_directory)
-
-        missions = ["SUZAKU", "ISO-IR", "Chandra", "XMM-OM-OPTICAL", "XMM", "XMM-OM-UV", "HST-IR", "Herschel",
-                    "Spitzer", "HST-UV", "HST-OPTICAL", "INTEGRAL"]
-
-        result = ESASkyClass.get_images(observation_ids=["100001010", "01500403", "21171", "0852000101", "0851180201",
-                                                         "0851180201", "n3tr01c3q", "1342247257", "30002561-25100",
-                                                         "hst_07553_3h_wfpc2_f160bw_pc", "ocli05leq", "88600210001"],
-                                        missions=missions, download_dir=download_directory)
-
-        for mission in missions:
-            file_path = os.path.join(download_directory, mission)
-            assert os.path.exists(file_path)
-            log.info("Checking {} data.".format(mission))
-            if mission.upper() == "HERSCHEL":
-                assert (isinstance(result[mission.upper()][0]["250"], HDUList))
-                assert (isinstance(result[mission.upper()][0]["350"], HDUList))
-                assert (isinstance(result[mission.upper()][0]["500"], HDUList))
-            else:
-                assert (isinstance(result[mission.upper()][0], HDUList))
-
-        result = None
-
-        shutil.rmtree(download_directory)
-
-    def test_esasky_get_spectra_obs_id(self):
-        download_directory = "ESASkyRemoteTest"
-        if not os.path.exists(download_directory):
-            os.makedirs(download_directory)
-
+    @pytest.mark.parametrize("mission, obsid",
+                             zip(["SUZAKU", "ISO-IR", "Chandra", "XMM-OM-OPTICAL",
+                                  "XMM", "XMM-OM-UV", "HST-IR", "Herschel",
+                                  "Spitzer", "HST-UV", "HST-OPTICAL", "INTEGRAL"],
+                                 ["100001010", "01500403", "21171", "0852000101",
+                                  "0851180201", "0851180201", "n3tr01c3q", "1342247257",
+                                  "30002561-25100", "hst_07553_3h_wfpc2_f160bw_pc", "ocli05leq", "88600210001"]))
+    def test_esasky_get_images_obs_id(self, tmp_path, mission, obsid):
+        result = ESASky.get_images(observation_ids=obsid,
+                                   missions=mission, download_dir=tmp_path)
+
+        assert (tmp_path / mission).exists()
+        if mission.upper() == "HERSCHEL":
+            assert (isinstance(result[mission.upper()][0]["250"], HDUList))
+            assert (isinstance(result[mission.upper()][0]["350"], HDUList))
+            assert (isinstance(result[mission.upper()][0]["500"], HDUList))
+        else:
+            assert (isinstance(result[mission.upper()][0], HDUList))
```

Comment:
```suggestion
            assert isinstance(result[mission.upper()][0]["250"], HDUList)
            assert isinstance(result[mission.upper()][0]["350"], HDUList)
            assert isinstance(result[mission.upper()][0]["500"], HDUList)
        else:
            assert isinstance(result[mission.upper()][0], HDUList)
```

Reply:
Good catch!  This usually drives me nuts, but I missed it here.  😄